src/containers/SearchPage/components/form/SearchForm.tsx

```diff
@@ -40,7 +44,15 @@ export interface SearchParams {
   "filter-inactive"?: boolean;
 }
 
-export const parseSearchParams = (locationSearch: string): SearchParams => {
+export type SearchParamsBody = IDraftConceptSearchParams &
+  IDraftSearchParams &
+  IImageSearchParams &
+  IAudioSearchParams &
+  ISeriesSearchParams;
+
+type ReturnType<T> = T extends true ? SearchParamsBody : SearchParams;
```

Comment:
Kanskje gi `T` et litt mer beskrivende navn her?

Reply:
Nå har ikke jeg så mye imot `T` her. Men hvis det ikke er for langt, kunne `ResourceSearchParams` vært noe; slik?

```suggestion
type ReturnType<ResourceSearchParams> = ResourceSearchParams extends true ? SearchParamsBody : SearchParams;
```